frontend/src/components/ui/BuySellMenu.jsx

```diff
@@ -84,10 +82,10 @@ export const BuySellMenu = ({game, selectedTower, setSelectedTower}) => {
 
                 const tower = new towerType.create(x, y, z, TowerStatus.PENDING);
                 game.addTower(tower);
-                gameMap.addTower(x, y);
+                gameMap.addTower(x, y, tower);
                 setUndoStack(prevStack => [...prevStack, {x, y, z, label: 'BuildTower', price: towerType.price}]);
                 game.setPath();
-                mouseInput.addClickCallback(SELL, (x, y, _z) => {
+                mouseInput.addClickCallback(SELECT, (x, y, _z) => {
```

Comment:
Huh, how did this get in here, isn't it a duplicate of the callback set above?

Reply:
It looks like there are two times `mouseInput.addClickCallback` is set to `SELECT`, when the game enters DEFEND phase, and after building (because the click callback has to be changed from `BUILD`)